src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -53,6 +56,271 @@ pub enum LinearActuatorMode {
     ClosedCircuitDamping,
 }
 
+#[derive(PartialEq, Copy, Clone)]
+struct VariableSpeedPump {
+    speed: LowPassFilter<AngularVelocity>,
+
+    is_powered: bool,
+    should_activate_electrical_mode: bool,
+    powered_by: ElectricalBusType,
+    consumed_power: Power,
+}
+impl VariableSpeedPump {
+    // Coefficient to convert hyd power in elec power 1.1 means we lose 10% efficiency from the pump
+    const HYD_TO_ELEC_POWER_EFFICIENCY: f64 = 1.15;
+    const MIN_STATIC_POWER_CONSUMPTION_WATT: f64 = 150.;
+
+    const FLOW_CONSTANT_RPM_CUBIC_INCH_TO_GPM: f64 = 231.;
+    const DISPLACEMENT_CU_IN: f64 = 0.214;
+    const NOMINAL_MAX_PRESSURE_PSI: f64 = 5000.;
+
+    const LOW_PASS_RPM_TRANSIENT_TIME_CONSTANT: Duration = Duration::from_millis(100);
+
+    pub fn new(powered_by: ElectricalBusType) -> Self {
+        Self {
+            speed: LowPassFilter::<AngularVelocity>::new(
+                Self::LOW_PASS_RPM_TRANSIENT_TIME_CONSTANT,
+            ),
+            is_powered: false,
+            should_activate_electrical_mode: false,
+            powered_by,
+            consumed_power: Power::default(),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        actuator_flow: VolumeRate,
+        actuator_pressure: Pressure,
+        controller: &impl ElectroHydrostaticPowered,
+    ) {
+        self.should_activate_electrical_mode = controller.should_activate_electrical_mode();
+
+        let new_speed = if self.is_active() {
+            AngularVelocity::new::<revolution_per_minute>(
+                actuator_flow.get::<gallon_per_minute>()
+                    * Self::FLOW_CONSTANT_RPM_CUBIC_INCH_TO_GPM
+                    / Self::DISPLACEMENT_CU_IN,
+            )
+        } else {
+            AngularVelocity::default()
+        };
+
+        self.update_power_consumed(actuator_flow, actuator_pressure);
+        self.speed.update(context.delta(), new_speed);
+    }
+
+    fn update_power_consumed(&mut self, actuator_flow: VolumeRate, actuator_pressure: Pressure) {
+        let current_fluid_power = Power::new::<watt>(
+            actuator_pressure.get::<pascal>() * actuator_flow.get::<cubic_meter_per_second>(),
+        );
+
+        self.consumed_power = if self.is_active() {
+            (current_fluid_power * Self::HYD_TO_ELEC_POWER_EFFICIENCY)
+                .max(Power::new::<watt>(Self::MIN_STATIC_POWER_CONSUMPTION_WATT))
+        } else {
+            Power::default()
+        };
+    }
+
+    fn is_active(&self) -> bool {
+        self.should_activate_electrical_mode && self.is_powered
+    }
+
+    fn max_available_pressure(&self, accumulator_pressure: Pressure) -> Pressure {
+        if self.is_active() && accumulator_pressure.get::<psi>() > 100. {
+            Pressure::new::<psi>(Self::NOMINAL_MAX_PRESSURE_PSI)
+        } else {
+            Pressure::default()
+        }
```

Comment:
Hmm, does it make sense to use a fixed threshold cutoff here? With this, it looks like the backup would be immediately available as long as there's any accumulator pressure.

Don't get me wrong, you probably still get some interesting dynamics, but I wonder if plan on going even more in-depth with this in the future 😄.

Reply:
Well we could map max pressure on accumulator pressure, but keep in mind accumulator in those actuators is only part of the low pressure side. So it is in fact the equivalent of the air pressurisation of our hyd reservoirs. So a low pressure here is in fact a fluid level indication basically. So we are already on a super edge case where the actuator would probably declare a fault and do not work at all anyway..... We can still add a map at some point if we feel there's a failure scenario for it to be noticeable